src/AppInstallerRepositoryCore/Rest/RestSourceFactory.cpp

```diff
@@ -0,0 +1,80 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+#pragma once
+#include "pch.h"
+#include "RestSourceFactory.h"
+#include "RestClient.h"
+#include "RestSource.h"
+
+using namespace std::string_literals;
+using namespace std::string_view_literals;
+
+namespace AppInstaller::Repository::Rest
+{
+	namespace
+	{
+		// The base class for data that comes from a rest based source.
+		struct RestSourceFactoryBase : public ISourceFactory
+		{
+			std::shared_ptr<ISource> Create(const SourceDetails& details, IProgressCallback& progress) override final
+			{
+				THROW_HR_IF(E_INVALIDARG, details.Type != RestSourceFactory::Type());
+				return CreateInternal(details, progress);

```

Comment:
No need to keep the pattern since there isn't actually a split based on the runtime here.  Move CreateInternal insides here.